stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/database/converter/ChannelInfoConverter.kt

```diff
@@ -0,0 +1,18 @@
+package io.getstream.chat.android.offline.repository.database.converter
+
+import androidx.room.TypeConverter
+import com.google.gson.reflect.TypeToken
+import io.getstream.chat.android.client.models.ChannelInfo
+import io.getstream.chat.android.offline.gson
+
+internal class ChannelInfoConverter {
```

Comment:
Why you decided to convert it to string and back? `ChannelInfo` is a flat value object (aka POJO/POKO). The room provides built-in mechanism for such use-case [Embeded objects](https://developer.android.com/reference/android/arch/persistence/room/Embedded).
```kotlin
internal data class MessageInnerEntity(
    @PrimaryKey
    val id: String,
    val cid: String,
    ....,
    @Embedded
    val channelInfo: ChannelInfoEntity
 }
 ```
 
 Here `ChannelInfoEntity` is a full-copy of `ChannelInfo`